src/main/java/com/google/api/codegen/gapic/SurfaceGapicProvider.java

```diff
@@ -0,0 +1,177 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.gapic;
+
+import com.google.api.codegen.ApiConfig;
+import com.google.api.codegen.GeneratedResult;
+import com.google.api.codegen.InputElementView;
+import com.google.api.codegen.java.direct.JavaClass;
+import com.google.api.codegen.java.surface.JavaSurface;
+import com.google.api.codegen.java.surface.JavaXApi;
+import com.google.api.codegen.languagedirect.LanguageDirectSnippetSetRunner;
+import com.google.api.codegen.surface.SurfaceSnippetSetRunner;
+import com.google.api.codegen.transformer.ModelToJavaDirectTransformer;
+import com.google.api.codegen.transformer.ModelToJavaSurfaceTransformer;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Model;
+import com.google.api.tools.framework.model.stages.Merged;
+import com.google.api.tools.framework.snippet.Doc;
+import com.google.common.base.Strings;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import javax.annotation.Nullable;
+
+public class SurfaceGapicProvider implements GapicProvider<Interface> {
```

Comment:
This currently is has a lot of Java-specific references for appearing to be language-agnostic code.